src/renderer/App.tsx

```diff
@@ -99,41 +101,45 @@ const processNewPdfs = async (pdfRootDir, nodes) => {
     );
   });
 
-  // const autograbNodes = pdfDirs.map((dir, ix) => {
-  //   return makeAutograbNode(
-  //     dir,
-  //     { dir },
-  //     { x: 50 + ix + Math.random() * 100, y: 50 + ix * Math.random() * 100 }
-  //   );
-  // });
-
   const allNodeIds = Object.keys(nodes);
 
   const newPubs = pdfNodes.filter(pdfNode => !allNodeIds.includes(pdfNode.id)); //filter out nodes that exists
-  // const newAutograbs = autograbNodes.filter(
-  //   autograbNode => !allNodeIds.includes(autograbNode.id)
-  // ); //filter out nodes that exists
-
-  // // add links btw nodes of type auto-grab and nodes of pdf.publication
-  // let newLinks = [];
-  // for (let i = 0; i < newPubs.length; i++) {
-  //   const linkToPdf = makeLink(newPubs[i].id, newAutograbs[i].id, {
-  //     type: "more"
-  //   });
-  //   newLinks.push(linkToPdf);
-  //   // assert each paper corresponds to one autograb node and idx are the same(for now)
-  // }
-
-  // concatenate nodes of type auto-grab and nodes of pdf.publication
-  // let newNodes = [] as aNode[];
-  // const nodesArray = newPubs.concat(autograbNodes);
-  // for (let i = 0; i < nodesArray.length; i++) {
-  //   newNodes.push(nodesArray[i]);
-  // }
-
-  // return new nodes and links batch to be added in Redux
-  // return { newNodes: newNodes };
-  return newPubs;
+
+  if (!featureToggles.showAutoGrab) {  // do not show auto-grab, return directly
+    return newPubs;
+  } else {
+    const autograbNodes = pdfDirs.map((dir, ix) => {
```

Comment:
This whole thing can be it's own function in Autograb.tsx

Reply:
under the else that is

Reply:
It could be, I will do it.

Reply:
Extracted into AutoGrab.tsx. 

Semantically, being scattered also makes sense. Since the block of `const autograbNodes = ... ` has a similar logic to the preceding block of `const pdfNodes = ... `.